plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -278,6 +278,80 @@ export const createSessionRecordingEvent = (
 
     return data
 }
+
+export interface SummarizedSessionRecordingEvent {
+    uuid: string
+    timestamp: string
+    team_id: number
+    distinct_id: string
+    session_id: string
+    first_url: string | undefined
+    click_count: number
+    keypress_count: number
+    mouse_activity_count: number
+}
+
+interface RRWebEventSummaryData {
+    href?: string
+    source?: number
+}
+
+interface RRWebEventSummary {
+    timestamp: number
+    type: number
+    data: RRWebEventSummaryData
+}
+
+export const createSessionReplayEvent = (
+    uuid: string,
+    team_id: number,
+    distinct_id: string,
+    timestamp: DateTime,
+    ip: string | null,
+    properties: Properties
+) => {
+    const timestampString = castTimestampOrNow(timestamp, TimestampFormat.ClickHouse)
+
+    const eventsSummaries: RRWebEventSummary[] = properties['$snapshot_data']?.['events_summary'] || []
+
+    if (eventsSummaries.length === 0) {
+        return null
+    }
+
+    let clickCount = 0
+    let keypressCount = 0
+    let mouseActivity = 0
+    let url: string | undefined = undefined
+    eventsSummaries.forEach((eventSummary: RRWebEventSummary) => {
+        if (eventSummary.type === 3) {
+            mouseActivity += 1
+            if (eventSummary.data?.source === 2) {
+                clickCount += 1
+            }
+            if (eventSummary.data?.source === 5) {
+                keypressCount += 1
+            }
+        }
+        if (!!eventSummary.data?.href?.trim().length && url === undefined) {
+            url = eventSummary.data.href
+        }
+    })
+
+    const data: SummarizedSessionRecordingEvent = {
+        uuid,
+        team_id: team_id,
+        distinct_id: distinct_id,
+        session_id: properties['$session_id'],
+        timestamp: timestampString,
```

Comment:
Not sure if it is just still to come but I assume this will be changed to `first_timestamp` and `last_timestamp`?

Reply:
Realised one confusion here - we aggregate to "first and last" in the clickhouse stuff _but_ we need to have at the individual event level _also_ first and last timestamps (first being the earliest timestamp in the list of event_summaries and last being the latest timestamp). This is because one "event" might contain 30 seconds of rrweb events so we need to save both values.

Reply:
🤯

Reply:
🤦

Reply:
done and done